crates/client/src/client.rs

```diff
@@ -287,40 +324,59 @@ impl FuelClient {
         Vars: serde::Serialize,
         ResponseData: serde::de::DeserializeOwned + 'static,
     {
-        let mut operation_json = serde_json::to_value(&q)?;
-        let operation_object = operation_json
-            .as_object_mut()
-            .expect("Graphql operation is a valid json object");
-        operation_object.insert(
-            "extensions".to_owned(),
-            serde_json::to_value(self.extensions.clone())?,
-        );
-
-        println!("{}", serde_json::to_string_pretty(&operation_json).unwrap());
-        let request_builder = self
+        let required_fuel_block_height = self.required_block_height();
+        let fuel_operation = FuelOperation::new(q, required_fuel_block_height);
+        let response = self
             .client
             .post(self.url.clone())
-            .headers(self.headers.clone())
-            .header(CONTENT_TYPE, "application/json")
-            .body(serde_json::to_string(&operation_json)?);
-
-        let response = request_builder
-            .send()
-            .await
-            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?
-            .text()
+            .run_fuel_graphql(fuel_operation)
             .await
             .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
 
-        println!("Response: {}", response);
-        let response = serde_json::from_str(&response)?;
-        Self::decode_response(response)
+        let inner_required_height = match &self.require_height {
+            ConsistencyPolicy::Auto { height } => Some(height.clone()),
+            _ => None,
+        };
+
+        Self::decode_response(response, inner_required_height)
     }
 
-    fn decode_response<R>(response: GraphQlResponse<R>) -> io::Result<R>
+    fn decode_response<R, E>(
+        response: FuelGraphQlResponse<R, E>,
+        inner_required_height: Option<Arc<Mutex<Option<BlockHeight>>>>,
+    ) -> io::Result<R>
```

Comment:
While this is okay, I think it would be cleaner to separate the height extraction to a separate `extract_required_height` function.